Devscord.DiscordFramework/Framework/Commands/Services/BotCommandsMatchingService.cs

```diff
@@ -8,82 +11,78 @@ namespace Devscord.DiscordFramework.Framework.Commands.Services
 {
     public class BotCommandsMatchingService
     {
+        private readonly Regex _exTime = new Regex(@"\d+(ms|d|h|m|s)", RegexOptions.IgnoreCase | RegexOptions.Compiled);
+        private readonly Regex _exUserMention = new Regex(@"\<@!\d+\>", RegexOptions.Compiled);
+        private readonly Regex _exChannelMention = new Regex(@"\<#\d+\>", RegexOptions.Compiled);
 
-        private readonly Regex _exTime = new Regex(@"\d+(h|m|s)", RegexOptions.IgnoreCase | RegexOptions.Compiled);
-        private readonly Regex _exUserMention = new Regex(@"\<@[^\d]*\d+\>", RegexOptions.Compiled);
-        private readonly Regex _exChannelMention = new Regex(@"\<#[^\d]*\d+\>", RegexOptions.Compiled);
-
-        public bool IsMatchedWithCommand(DiscordRequest request, BotCommandTemplate template)
+        public bool IsNormalCommand(DiscordRequest request, BotCommandTemplate template)
         {
             if (!request.IsCommandForBot)
             {
                 return false;
             }
-            if (request.Name.ToLowerInvariant() != template.NormalizedCommandName)
-            {
-                return false;
-            }
-            if (!this.CompareArgumentsToProperties(request.Arguments.ToList(), template.Properties.ToList()))
+            return request.Name.ToLowerInvariant() == template.NormalizedCommandName;
+        }
+
+        public bool IsMatchedWithNormalCommand(BotCommandTemplate template, IEnumerable<DiscordRequestArgument> arguments)
+        {  
+            var requiredProperties = template.Properties.Where(x => !x.IsOptional);
+            if (requiredProperties.Any(property => !arguments.Any(arg => property.Name.ToLowerInvariant() == arg.Name?.ToLowerInvariant())))
```

Comment:
```suggestion
            if (requiredProperties.Any(property => arguments.All(arg => property.Name.ToLowerInvariant() != arg.Name?.ToLowerInvariant())))
```

Reply:
to nie ma sensu, bo wtedy nawet jeśli komenda będzie poprawna to zostanie uznana za błędną ponieważ wymagane property musi być w Arguments a tutaj w sugestii u ciebie to musi go nie być.

Reply:
to jest z punktu logiki dokładnie to samo, tylko inaczej zapisane
u ciebie:
sprawdza czy jakikolwiek name'y są takie same, jeśli tak to Any zwraca true, ! odwraca, więc ostatecznie wychodzi false
u mnie:
sprawdza czy wszystkie name'y są różne (zakładamy że ten jeden jest taki sam) to zwraca false
czyli to samo co u ciebie
jeśli wszystkie są różne:
u ciebie:
any zwraca false, ! odwraca i ostatecznie jest true
u mnie:
all zwraca true (bo wszystkie są różne) i ostatecznie zostaje to true